core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushLimitThroughProject.java

```diff
@@ -67,14 +68,18 @@ public Result apply(LimitNode parent, Captures captures, Context context)
             return Result.empty();
         }
 
-        // for a LimitNode without ties, simply reorder the nodes
-        if (!parent.isWithTies()) {
+        // for a LimitNode without ties and input order sensitivity, simply reorder the nodes
+        if (!parent.isWithTies() && !parent.isOrderSensitive()) {
             return Result.ofPlanNode(transpose(parent, projectNode));
         }
 
         // for a LimitNode with ties, the tiesResolvingScheme must be rewritten in terms of symbols before projection
         SymbolMapper.Builder symbolMapper = SymbolMapper.builder();
-        for (Symbol symbol : parent.getTiesResolvingScheme().get().getOrderBy()) {
+        Set<Symbol> symbolsForRewrite = ImmutableSet.<Symbol>builder()
+                .addAll(parent.getPreSortedInputs())
+                .addAll(parent.getTiesResolvingScheme().map(OrderingScheme::getOrderBy).orElse(ImmutableList.of()))
```

Comment:
I think `parent.getTiesResolvingScheme().map(OrderingScheme::getOrderBy).orElse(ImmutableList.of())` is needed instead of `parent.getTiesResolvingScheme().get().getOrderBy()` because we can reach here when `isWithTies` is `false` but `getPreSortedInputs` is not empty.
I created the intermediate Set because `SymbolMapper.builder` is using an `ImmutableMap.Builder` internally which will throw exception on encountering duplicate key. Can we assume that pre-sorted inputs and tiesResolvingScheme won't have overlap in symbols ?